Sources/NIO/NIOCloseOnErrorHandler.swift

```diff
@@ -0,0 +1,27 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+
+/// A `ChannelInboundHandler` that closes the channel when an error is caught
+public final class NIOCloseOnErrorHandler: ChannelInboundHandler {
+
+    public typealias InboundIn = NIOAny
+    
+    /// Initialize a `NIOCloseOnErrorHandler`
+    public init() {}
+    
+    public func errorCaught(context: ChannelHandlerContext, error: Error) {
+        context.close(promise: nil)

```

Comment:
Here's a question: should this forward the error on? @weissi?

Reply:
@Lukasa I think I agree,  yes, don't think this could do anything harmful. In most cases absolutely nothing will happen (because this will likely be the last handler) but in certain cases it might be better to give the user visibility.

Reply:
So, forward then close? I’m guessing the order matters here

Reply:
@Lukasa @weissi tried something in 061f1632, is that the right approach?

Reply:
yes, that works!

Reply:
Perfect, squashed into one commit!